Revive scatterplot: re-add scatterplot to Deployments page

rmars opened the pull request:
Attempt to revive the scatterplot we once had on the deployments overview page.
Tried to make some UI improvements to address previous problems: 
- added a hover bar and tooltip that displays all of the nodes under the bar, in descending order of successRate (to correspond with their order in the chart)
- the tooltip looked weird in the empty state so I also added the max/min latencies observed there

Also cleans up the Deployments page a little when there are not any "least healthy deployments".

The colours and the whole thing could certainly use further tweaking, which I will keep doing.

![screenshot](https://user-images.githubusercontent.com/549258/34855852-d0c1393a-f6f6-11e7-9480-61f370b24ae5.png)

siggy commented on web/app/js/components/ScatterPlot.jsx:
```diff
@@ -128,49 +162,58 @@ export default class ScatterPlot extends React.Component {
+    this.overlay
+      .on("mousemove", () => {
+        let currXPos = d3.mouse(d3.select("rect").node())[0];
+        this.verticalHighlight.attr("transform", "translate(" + currXPos + ", 0)");
```
i noticed when i move the mouse left, the vertical highlight smoothly follows, but moving the mouse right, the highlight stays put until i hit its right edge, and then it jumps a full width?

rmars replied:
whooooaaaa that's interesting. lemme look into it.

rmars commented:
Yeah, re: the labels, I removed the per-node label that used to appear in the top RH corner. maybe I could add some labels to the overlay.....

rmars commented:
Took a stab ad adding the nodenames to the overlay as a tooltip. lmk what you think!

rmars commented:
hmmmmmm that shoudn't happen :| let me investigate

rmars commented:
Fixed the problem above: when there was only one datapoint (e.g. at n), the scale's range was [n, n] and getNearbyDatapoints always returned all the datapoints. Fixed this to check for datapoints that are actually near the mouse in this case.

rmars commented:
It definitely is, but the reason I wanted to have all of the labels attached to the vertical highlight was in the case of many nodes crowded into the same space (a common problem plaguing the first iteration). If they're attached to the circles, it's a lot harder to make sure the labels won't overlap. 

For cases like the above, it's still confusing which is which (they're always ordered by descending success rate, so they will correspond vertically).  I could make more of an attempt to detect whether we're going to be labelling a lot of nodes, and if we aren't, we can align each label horizontally with the node it's labelling. I didn't want to preemptively spend a lot of time on it, but I think it's not too much more work.

rmars commented:
Ok, we now try to align each label with the node its labelling, except if the labels would overlap too much.
